app/controllers/surveys_controller.rb

```diff
@@ -8,4 +8,11 @@ class SurveysController < ApplicationController
   def index
     @surveys = (administrator_signed_in? ? Survey.all : Survey.active).descending
   end
+
+  def qrcode
+    @survey = Survey.find(params[:id])
+    @qr_code = QrCodeService.new(@survey.short_url)
+
+    render 'qrcode'
```

Comment:
Not needed

Reply:
What is not needed? The qr_code is non existent otherwise

Reply:
The render call. Rails will render a view which is named like your action.